src/sentry/monitors/logic/mark_failed.py

```diff
@@ -145,18 +146,16 @@ def mark_failed_threshold(failed_checkin: MonitorCheckIn, failure_issue_threshol
 
         starting_checkin = previous_checkins[0]
 
-        # for new incidents, generate a uuid as the fingerprint. This is
-        # not deterministic of any property of the incident and is simply
-        # used to associate the incident to it's event occurrences
-        fingerprint = uuid.uuid4().hex
-
-        MonitorIncident.objects.create(
+        incident, _ = MonitorIncident.objects.get_or_create(
             monitor=monitor_env.monitor,
             monitor_environment=monitor_env,
```

Comment:
I think we need to be explicit about the `resolved_checkin=None` here

Reply:
Yeah sorry hadn't had a chance to update. Updated now.